### ISSUE TITLE
--contained-run keeps asking for git password

### ISSUE THREAD

**Issue by JanStreffing** [esm-tools/esm_tools#1008]
On aleph I tried setting up a production run. It keeps asking me for my password on the ssh key I use for github. I typed it in correctly typed every time. I know it was correct, because after every correct pw it takes about 5 seconds to ask again, while when I entered the wrong one it takes 0.5s. After a while I gave up and started with --open-run.

```
esm_runscripts  TCo1279-DART-1Y.yaml -e TCo1279-DART-1Y --contained-run -c
Building virtual env, please be patient (this takes about 3 minutes)...
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
Enter passphrase for key '/home/awiiccp2/.ssh/id_ed25519_git': 
```

I'm not sure what further information to provide. Here is a folder in which you can see how for the esm_tools got:
`/scratch/awiiccp2/does_not_contain/.venv_esmtools`
Here is the runscript:
`/home/awiiccp2/esm_tools/runscripts/awicm3/frontiers-xios/TCo1279-DART-1Y.yaml`

I've never been prompted to enter my password here before, so I'm not quite sure what's going on.

**Comment by mandresm** [comment]
The answer is that it's probably installing multiple packages, not only ESM-Tools but also the oasis copying plugin, preproces0s plugin etc. In the end we are simply doing `git clone`s so that would be kind of expected if `git clone https://github.com/esm-tools/esm_tools` asks you for a password. Can you try the following?

```
eval `ssh-agent`
ssh-add /home/awiiccp2/.ssh/id_ed25519_git
try to build your venv again and see if you only need one password now
```

If that's the case we can maybe add `eval `ssh-agent`` to that part of the virtual environment builder and also the `ssh-add` call

**Comment by JanStreffing** [comment]
That solved it! Thank you. How did you find this?

**Comment by mandresm** [comment]
How do I found about ssh-eval or that the problem was that it's trying to install esm_tools plus many plugins?

About the first one, I learned about reading the docs of HLRN system. It's quite cool to don't have to always introduce your ssh-key password.

About the second part, I think I was the one introducing the plugins automatic installation into the venv so it jumped to my mind inmediatelly.

Note that you'll see this problem ONLY if you use ssh-key by default to clone from GitHub. Then it's ssh who's asking for a password to your ssh-key. If you'd try to clone a public repository from GitHub (which is the case) without ssh-key you'd need no password and that's why you only see this problem now.

Feel free to close it, if you consider it sorted.
